task8/src/main/java/AdCollection/Ad.java

```diff
@@ -0,0 +1,145 @@
+package AdCollection;
+import org.json.simple.JSONObject;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class Ad {
+    private int id;
+    private String description;
+    private String createdAt;
+    private String link;
+    private String vendor;
+    private String photoLink;
+    private List<String> hashTags;
+    private String discount;
+    private String validUntil;
+    private Double rating;
+    private List<String> reviews;
+
+
+    public Ad(int id, String description, String createdAt, String link, String vendor, List<String> hashTags, String discount, String validUntil, String photoLink) {
+        this.id = id;
+        this.description = description;
+        this.createdAt = createdAt;
+        this.link = link;
+        this.vendor = vendor;
+        this.hashTags = hashTags != null ? new ArrayList<>(hashTags) : null;
+        this.discount = discount;
+        this.photoLink = photoLink;
+        this.validUntil = validUntil;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public void setId(int id) {
+        this.id = id;
```

Comment:
Semantically this is barely correct. ID is considered to be an immutable attribute of an entity, "once and for all"  Why does one need to re-assign an ID?